pkg/bicep/biceptoschema.go

```diff
@@ -0,0 +1,231 @@
+package bicep
+
+import (
+	"encoding/json"
+	"errors"
+	"fmt"
+	"reflect"
+	"slices"
+
+	bp "github.com/Checkmarx/kics/v2/pkg/parser/bicep"
+	"github.com/massdriver-cloud/airlock/pkg/schema"
+
+	orderedmap "github.com/wk8/go-ordered-map/v2"
+)
+
+type bicepParam struct {
+	TypeString    string             `json:"type"`
+	DefaultValue  interface{}        `json:"defaultValue"`
+	AllowedValues []interface{}      `json:"allowedValues"`
+	Metadata      bicepParamMetadata `json:"metadata"`
+	MinLength     *uint64            `json:"minLength"`
+	MinValue      *uint64            `json:"minValue"`
+	MaxLength     *uint64            `json:"maxLength"`
+	MaxValue      *uint64            `json:"maxValue"`
+}
+
+type bicepParamMetadata struct {
+	Description string                 `json:"description"`
+	Metadata    map[string]interface{} `json:"metadata"`
+}
+
+func BicepToSchema(valuesPath string) (string, error) {
+	// using the github.com/Checkmarx/kics parser since he already did the heavy lifting to parse a bicep template
+	parser := bp.Parser{}
+
+	params := new(schema.Schema)
+	params.Type = "object"
+	params.Properties = orderedmap.New[string, *schema.Schema]()
+	params.Required = []string{}
+
+	doc, _, err := parser.Parse(valuesPath, nil)
+	if err != nil {
+		return "", err
+	}
+
+	for name, value := range doc[0]["parameters"].(map[string]interface{}) {
+		bicepParam := bicepParam{}
+
+		// marshall to json and unmarshall into custom struct to make bicep param easier to access
+		bytes, err := json.Marshal(value)
+		if err != nil {
+			return "", err
+		}
+		err = json.Unmarshal(bytes, &bicepParam)
+		if err != nil {
+			return "", err
+		}
+
+		property := new(schema.Schema)
+		property.Title = name
+		property.Description = bicepParam.Metadata.Description
+
+		err = parseBicepParam(property, bicepParam)
+		if err != nil {
+			return "", err
+		}
+
+		params.Properties.Set(name, property)
+		params.Required = append(params.Required, name)
+	}
+	// sorting this here just to help with testing. The order doesn't matter, but to our test suite it does.
+	slices.Sort(params.Required)
+
+	out, err := json.Marshal(params)
+	if err != nil {
+		return "", err
+	}
+
+	return string(out), nil
+}
+
+func parseBicepParam(sch *schema.Schema, bicepParam bicepParam) error {
+	switch bicepParam.TypeString {
+	case "int":
+		return parseIntParam(sch, bicepParam)
+	case "bool":
+		return parseBoolParam(sch, bicepParam)
+	case "string":
+		return parseStringParam(sch, bicepParam, false)
+	case "secureString":
+		return parseStringParam(sch, bicepParam, true)
+	case "array":
+		return parseArrayParam(sch, bicepParam)
+	case "object", "secureObject":
+		return parseObjectParam(sch, bicepParam)
+	default:
```

Comment:
are types like numbers, floats not supported?

Reply:
Yeah, bicep only support integers currently. https://learn.microsoft.com/en-us/azure/azure-resource-manager/bicep/data-types#integers

> Floating point, decimal or binary formats aren't currently supported.